test/topology_experimental_raft/test_topology_upgrade.py

```diff
@@ -12,16 +12,19 @@
 from test.pylib.rest_client import HTTPError
 from test.pylib.manager_client import ManagerClient
 from test.pylib.util import wait_for_cql_and_get_hosts
-from test.topology.util import log_run_time, wait_until_topology_upgrade_finishes, \
+from test.topology.util import log_run_time, wait_until_last_generation_is_in_use, wait_until_topology_upgrade_finishes, \
         wait_for_cdc_generations_publishing, check_system_topology_and_cdc_generations_v3_consistency, \
         start_writes_to_cdc_table
 
 
 @pytest.mark.asyncio
 @log_run_time
-async def test_topology_upgrade_basic(request, manager: ManagerClient):
+async def test_topology_upgrade_basic(request, mode: str, manager: ManagerClient):
     # First, force the first node to start in legacy mode due to the error injection
-    cfg = {'error_injections_at_startup': ['force_gossip_based_join']}
+    cfg = {
```

Comment:
Instead of increasing `ring_delay`, you could increase `generation_leeway`. There is an error injection that sets it to 5 min, which is too large for this test. Other test requires 5 min though. However, you could replace this injection with the recently merged `inject_parameter`. I like increasing `generation_leeway` instead of `ring_delay` because it achieves exactly what we want without impacting the rest of the system. This new `inject_parameter` could also be useful in future tests. Of course, this suggestion is optional. It might be too much effort for too little reward.

Reply:
> It might be too much effort for too little reward.

I agree. It would be useful if more tests use it (for now only test_topology_upgrade and test_recovery_basic)